### PR TITLE
Refactor the user controller

### PR DESCRIPTION
**Pull request by ghost**
#### What does this PR do?
- Refactor user controller
- Write controller tests

#### Description of Task to be completed?
N/A

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -16,4 +16,18 @@ def require_user
       redirect_to root_path
     end
   end
+
+  def require_same_user(obj, path)
+    if current_user != obj && !current_user.admin?
```
Comment (robocopkaka) [inline app/controllers/application_controller.rb]: Probably best to change `obj` to something like `user`

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class UsersController < ApplicationController
-  before_action :set_user, only: [:edit, :update, :show, :destroy]
-  before_action :require_same_user, only: [:edit, :update, :destroy]
-  before_action :require_admin, only: [:destroy]
+  before_action :set_user, except: [:index, :new, :create]
+  before_action -> { require_same_user(@user, root_path) }, only: [:edit, :update]
```
Comment (robocopkaka) [inline app/controllers/users_controller.rb:3]: Any reason why you're using a Lambda here?

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,149 @@
+require "rails_helper"
```
Comment (robocopkaka) [inline spec/requests/users_spec.rb:1]: Minor observation: just realized that this is a regular Rails app and not an API one so writing feature specs with Capybara should be the way to go. Doesn't have to be done in this PR

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (ghost) [reply app/controllers/users_controller.rb:3]: The `require_same_user` method takes in 2 variables, to use it in a `before_action` I need to put it in a block

**Pull request merged**